pyproject.toml

```diff
@@ -1,6 +1,49 @@
 [build-system]
-build-backend = "setuptools.build_meta:__legacy__"
+build-backend = "setuptools.build_meta"
 requires = [
     "setuptools",
     "wheel",
 ]
+
+[project]
+name = "cpplint"
+dynamic = ["version", "optional-dependencies"]
+maintainers = [
+    { name = "Aaron Liu", email = "aaronliu0130@gmail.com" },
+    { name = "John Vandenberg", email = "jayvdb@gmail.com" }
+]
+authors = [
+    { name = "Google Inc."},
+    { name = "Thibault Kruse"},
+    { name = "Andrew Davis", email = "theandrewdavis@gmail.com" }
+]
+description="Check C++ files configurably against Google's style guide"
+readme = "README.rst"
+requires-python = ">=3.8"
+keywords = ["lint", "cpp", "c++", "google style"]
+license = { text = "BSD-3-Clause" }
+classifiers = [
+    "Development Status :: 5 - Production/Stable",
+    "Intended Audience :: Developers",
+    "License :: OSI Approved :: BSD License",
+    "Natural Language :: English",
+    "Programming Language :: Python :: 3 :: Only",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3.12",
+    "Programming Language :: Python :: 3.13",
+    "Programming Language :: Python :: 3.14",
+    "Topic :: Software Development :: Quality Assurance",
+]
+
+[tool.setuptools.dynamic]
+version = {attr = "cpplint.__VERSION__"}
+
+[tool.setuptools.dynamic.optional-dependencies]
```

Comment:
Why store requirements in separate files?  Bringing them into pyproject.toml streamlines project config.

Why not unify `test` and `dev` into just `dev`?

Reply:
It just felt more understandable to me for aesthetic reasons. No objections if you merge them into the TOML.

As for test/dev unification... good question. This does seem like excessive categorization. Please go ahead and make the commits to this PR for unifying the extras (and updating the documentation to reflect that).

Reply:
test and dev have now been unified.